src/features/images/tasks/delete-registry-images.ts

```diff
@@ -0,0 +1,176 @@
+import { permissions, sbvrUtils, tasks } from '@balena/pinejs';
+import _ from 'lodash';
+import { generateToken } from '../../registry/registry.js';
+import {
+	ASYNC_TASK_DELETE_REGISTRY_IMAGES_BATCH_SIZE,
+	REGISTRY2_HOST,
+} from '../../../lib/config.js';
+import { requestAsync } from '../../../infra/request-promise/index.js';
+import { setTimeout } from 'node:timers/promises';
+
+const { api } = sbvrUtils;
+
+// For registry API requests
+const DELAY = 200;
+const RATE_LIMIT_DELAY_BASE = 1000;
+const RATE_LIMIT_RETRIES = 5;
+
+const schema = {
+	type: 'object',
+	properties: {
+		images: {
+			type: 'array',
+			items: {
+				type: 'array',
+				items: {
+					type: 'string',
+				},
+				maxItems: 2,
+				minItems: 2,
+			},
+		},
+	},
+	required: ['images'],
+};
```

Comment:
@joshbwlng just for my own information, is there any specific reason you chose this schema instead of

```suggestion
const schema = {
	type: 'object',
	properties: {
		images: {
			type: 'array',
			items: {
				type: 'object',
				properties: {
					repo: { type: 'string' },
					hash: { type: 'string' },
				},
				required: ['repo', 'hash'],
				additionalProperties: false,
			},
		},
	},
	required: ['images'],
};
```
where the properties have explicit naming? (slightly easier to read imo)